src/main/java/org/ibp/api/java/impl/middleware/study/StudyEntryServiceImpl.java

```diff
@@ -91,13 +117,60 @@ public StudyEntryDto replaceStudyEntry(final Integer studyId, final Integer entr
 			.replaceStudyEntry(studyId, entryId, gid, this.pedigreeService.getCrossExpansion(gid, this.crossExpansionProperties));
 	}
 
+	@Override
+	public List<StudyEntryDto> createStudyEntries(final Integer studyId,
+		final StudyEntryGeneratorBatchRequestDto studyEntryGeneratorBatchRequestDto, final String programUuid) {
+		this.studyValidator.validate(studyId, true);
+		if(studyEntryGeneratorBatchRequestDto.getListId() != null && studyEntryGeneratorBatchRequestDto.getListId() != 0) {
+			return this.createStudyEntries(studyId, studyEntryGeneratorBatchRequestDto.getListId());
+		}
+
+		//Validate EntryType
+		this.entryTypeValidator.validateEntryType(studyEntryGeneratorBatchRequestDto.getEntryTypeId(), programUuid);
+
+		final SearchCompositeDto<Integer, Integer> searchComposite = studyEntryGeneratorBatchRequestDto.getSearchComposite();
+		final BindingResult errors = new MapBindingResult(new HashMap<>(), LotGeneratorBatchRequestDto.class.getName());
+		this.searchCompositeDtoValidator.validateSearchCompositeDto(searchComposite, errors);
+		final List<Integer> gids = this.searchRequestDtoResolver.resolveGidSearchDto(searchComposite);
+		this.germplasmValidator.validateGids(errors, gids);
+
+		final List<Germplasm> germplasmList = this.germplasmDataManager.getGermplasmWithPreferredNameByGIDs(gids);
```

Comment:
It seems heavy-handed that we have to get the whole Germplasm object (Hibernate entity) for selected GIDs when we are only going to get the preferred name.Is it possible to use the existing `GermplasmDataManager#getPreferredNamesByGids` instead?